### ISSUE TITLE
Cluster deployments Helm v2 -> v3 migration made traefik crash with OOMKill

### ISSUE THREAD

**Issue by sylr** [traefik/traefik#7384]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

We have migrated all the deployments of a Kubernetes cluster from helm v2.x to v3.x (traefik itself is **not** deployed with Helm).

Helm v2 used to store Helm "releases" using **ConfigMaps**. In Helm v3 the "releases" are stored using **Secrets**.

The contents of theses new Secrets (we had around 700 of them) are quite large binaries (at least hundreds of KiB, but lesser than 1MiB which is kubernetes/etcd limitation) and it seems that Traefik downloads them all from Kubernetes and keeps them in memory (or at least it does not have time to discard them before it is killed).

FYI before we made the migration the memory limit for Traefik pods was 512Mi and we never had a problem. We had to increase it to 2Gi to make it work until we understood where the memory consumption was coming from.

For the record Helm v2 made the `nginx-ingress-controller` behave quite similarly (OOMkilled because it reached the limit) because it was downloading all the clusters ConfigMaps including all the ones storing Helm v2 releases. The following patch was merged to avoid downloading Helm related ConfigMaps any longer https://github.com/kubernetes/ingress-nginx/pull/5702. Note that this would not work as it for Traefik and Helm v3 Secrets as it seems that Helm v3 Secret labeling use `owner=helm` instead of `OWNER=TILLER`, i don't know if the filtering is case sensitive. 

### What did you expect to see?

I would have expected that traefik does not fetch all the secrets from the Kubernetes api-server, only the ones it needs.

### What did you see instead?

Traefik being killed because it consumes too much memory.

### Output of `traefik version`: (_What version of Traefik are you using?_)

[v2.2.1+sylr.1](https://github.com/sylr/traefik/commits/v2.2.1%2Bsylr.4)

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```toml
    [log]
      level = "INFO"
      format = "json"
    #[tracing]
    #  [tracing.instana]
    #    localAgentHost = "instana-agent.instana.svc.cluster.local."
    #    localAgentPort = 42699
    #    logLevel = "info"
    [accessLog]
      format = "json"
      [accessLog.fields]
        defaultMode = "keep"
      [accessLog.fields.names]
        ClientAddr = "drop"
        RequestAddr = "drop"
      [accessLog.fields.headers]
        [accessLog.fields.headers.names]
          "User-Agent" = "keep"
    [entryPoints.http]
      address = ":80"
    [entryPoints.http-internal]
      address = ":81"
    [entryPoints.https]
      address = ":443"
    [entryPoints.traefik]
      address = ":8080"
    [entryPoints.traefik-ext]
      address = ":8081"
    [providers.kubernetesIngress]
      ingressClass = "traefik-v2"
    [providers.kubernetesCRD]
    [providers.file]
      directory  = "/config/static"
      watch = true
    [api]
      insecure = true
      dashboard = true
      debug = true
    [metrics.prometheus]
      entryPoint = "traefik"
      buckets = [0.1,0.25,0.5,0.75,1.0,1.5,2.0]
```

### If applicable, please paste the log output in DEBUG level (`--log.level=DEBUG` switch)

```
(paste your output here)
```

**Comment by sylr** [comment]
We also tried to play with our configuration and add a label selector:

```
[providers.kubernetesCRD]
      labelSelector = "owner notin (helm)"
```

This did not had any impact on the memory consumption which sounds normal according to this code:

https://github.com/traefik/traefik/blob/86c099d6297bd7103bef8240e90f45492be56000/pkg/provider/kubernetes/crd/client.go#L375-L398

I did not think to add this selector to the `kubernetesIngress` provider but I suspect it wouldn't have any effect neither:

https://github.com/traefik/traefik/blob/86c099d6297bd7103bef8240e90f45492be56000/pkg/provider/kubernetes/ingress/client.go#L382-L399

**Comment by jbdoumenjou** [comment]
Hi @sylr ,

Could you provide a reproducible case with an official version of Traefik?

Thank you.

**Comment by jbdoumenjou** [comment]
Hi @sylr,

We were looking at your PR, and we have some questions regarding the use case.

You said that Traefik is not deployed using Helm but you have a lot of new secrets generated through helm storage [backend](https://helm.sh/docs/topics/advanced/#storage-backends), does it mean that you use the same namespace for all objects?

If you do so, could you have specific namespaces for Traefik and needed resources and then use the [namespaces](https://doc.traefik.io/traefik/providers/kubernetes-crd/#namespaces) watched by the Kubernetes provider?

**Comment by sylr** [comment]
No, I do not have only one namespace, I have dozens of them but they are all watched by traefik as they all contains ingresses and ingressroutes.

**Comment by traefiker** [comment]
Closed by #7562.
